config/m4/java.m4

```diff
@@ -0,0 +1,63 @@
+#
+# Copyright (C) Mellanox Technologies Ltd. 2001-2017.  ALL RIGHTS RESERVED.
+#
+# See file LICENSE for terms.
+#
+#
+#
+# Check for Java support
+#
+#
+java_happy="no"
+AC_ARG_WITH([java],
+            [AC_HELP_STRING([--with-java],
+                            [Compile Java UCX (default is NO).]
+                           )], [with_java=yes], [with_java=no])
+
+
+AS_IF([test "x$with_java" != xno],
+      [AS_IF([test "x$with_java" == xyes],
+             [AS_IF([test -n "$JAVA_HOME"],
+                    [],
+                    [
+                     AC_SUBST([JAVA], [$(readlink -f $(which java))])
```

Comment:
* readlink should be detected, here is an example https://github.com/CZ-NIC/libnetconf/blob/master/configure.in#L317

* s/which/type